elastic/kibana
# [Canvas] Adds references extract/inject for Workpads

## Summary
This PR adds inject/extract functionality for references on a Canvas Workpad.  

I introduce some workpad CRUD methods onto a request context so that they will be available on all of our routes using the appropriately scoped Saved Objects Client.  These methods also implement the reference inject/extract where needed.  

Because the inject/extract happens server side, we need to make sure that any function that is going to need extract is available server side as well, so this also required some changes to our embeddable functions to only use code that will run server side (it was previously importing some code from /public dirs and that would crash Kibana start up). 

This also touches a few other plugins to move around some of the previously mentioned imports that were in public dirs to make sure they are coming from common, so that they will work both client and server side.  

#### Testing
Here are the important areas to test 
- Saving/Reloading a Workpad with an embeddable (Map, Lens, Visualization)
- Saved Object Management Export/Import Workpad with embeddables
  - Verify SO Management shows Links to referenced Saved Objects
- Export Workpad as JSON.  Import Workpad through Canvas home screen. Verify that has correct references from SO Managmeent
